internal/api/grpc/user/v2/query_integration_test.go

```diff
@@ -0,0 +1,629 @@
+					resp := Tester.CreateHumanUserVerified(ctx, orgResp.OrganizationId, username)
+					request.UserId = resp.GetUserId()
+					return nil
+				},
+			},
+			want: &user.GetUserByIDResponse{
+				User: &user.User{
+					State:              user.UserState_USER_STATE_ACTIVE,
+					Username:           "",
+					LoginNames:         nil,
+					PreferredLoginName: "",
+					Type: &user.User_Human{
+						Human: &user.HumanUser{
+							Profile: &user.HumanProfile{
+								GivenName:         "Mickey",
+								FamilyName:        "Mouse",
+								NickName:          gu.Ptr("Mickey"),
+								DisplayName:       gu.Ptr("Mickey Mouse"),
+								PreferredLanguage: gu.Ptr("nl"),
+								Gender:            user.Gender_GENDER_MALE.Enum(),
+								AvatarUrl:         "",
+							},
+							Email: &user.HumanEmail{
+								IsVerified: true,
+							},
+							Phone: &user.HumanPhone{
+								Phone:      "+41791234567",
+								IsVerified: true,
+							},
+						},
+					},
+				},
+				Details: &object.Details{
+					ChangeDate:    timestamppb.Now(),
+					ResourceOwner: orgResp.OrganizationId,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			username := fmt.Sprintf("%d@mouse.com", time.Now().UnixNano())
+			err := tt.args.dep(tt.args.ctx, username, tt.args.req)
+			require.NoError(t, err)
+
+			var got *user.GetUserByIDResponse
+			for {
+				got, err = Client.GetUserByID(tt.args.ctx, tt.args.req)
+				if err == nil || (tt.wantErr && err != nil) {
+					break
+				}
+				select {
+				case <-CTX.Done():
+					t.Fatal(CTX.Err(), err)
+				case <-time.After(time.Second):
+					t.Log("retrying GetUserByID")
+					continue
+				}
+			}
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				tt.want.User.UserId = tt.args.req.GetUserId()
+				tt.want.User.Username = username
+				tt.want.User.PreferredLoginName = username
+				tt.want.User.LoginNames = []string{username}
+				if human := tt.want.User.GetHuman(); human != nil {
+					human.Email.Email = username
+				}
+				require.Equal(t, tt.want.User, got.User)
+				integration.AssertDetails(t, tt.want, got)
+			}
+		})
+	}
+}
+
+type userAttr struct {
+	UserID   string
+	Username string
+}
+
+func TestServer_ListUsers(t *testing.T) {
+	orgResp := Tester.CreateOrganization(IamCTX, fmt.Sprintf("ListUsersOrg%d", time.Now().UnixNano()), fmt.Sprintf("%d@mouse.com", time.Now().UnixNano()))
+	//userResp := Tester.CreateHumanUserVerified(IamCTX, orgResp.OrganizationId, fmt.Sprintf("%d@listusers.com", time.Now().UnixNano()))
+	type args struct {
+		ctx   context.Context
+		count int
+		req   *user.ListUsersRequest
+		dep   func(ctx context.Context, org string, usernames []string, request *user.ListUsersRequest) ([]userAttr, error)
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *user.ListUsersResponse
+		wantErr bool
+	}{
+		/* commented as permission don't work as expected
```

Comment:
Don't we introduce a security bug if we release the API then?

Reply:
This problem is related to the CTXs used here, with which I couldn't handle the permissions in the integration tests, would need to have a look at it again.

Reply:
I think it's worth from a security perspective to find a solution to define a context with certain permissions and having it tested